src/frontend/src/table/delete.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use api::v1::DeleteRequest as GrpcDeleteRequest;
+use common_query::Output;
+use futures::future;
+use meta_client::rpc::TableName;
+use snafu::ResultExt;
+use store_api::storage::RegionNumber;
+use table::requests::DeleteRequest;
+
+use crate::error::{JoinTaskSnafu, RequestDatanodeSnafu, Result};
+use crate::table::insert::to_grpc_columns;
+use crate::table::DistTable;
+
+impl DistTable {
+    pub(super) async fn dist_delete(&self, requests: Vec<GrpcDeleteRequest>) -> Result<Output> {
+        let regions = requests.iter().map(|x| x.region_number).collect();
+        let instances = self.find_datanode_instances(regions).await?;
+
+        let results = future::try_join_all(instances.into_iter().zip(requests.into_iter()).map(
+            |(instance, request)| {
+                common_runtime::spawn_write(async move {
+                    instance
+                        .grpc_delete(request)
+                        .await
+                        .context(RequestDatanodeSnafu)
+                })
+            },
+        ))
+        .await
+        .context(JoinTaskSnafu)?;
+
+        let affected_rows = results.into_iter().collect::<Result<Vec<_>>>()?;
```

Comment:
This might work https://doc.rust-lang.org/stable/std/result/enum.Result.html#impl-Sum%3CResult%3CU,+E%3E%3E-for-Result%3CT,+E%3E
```suggestion
        let affected_rows = results.into_iter().sum::<Result<u32>>()?;
```